scripts/dts/gen_defines.py

```diff
@@ -575,6 +579,18 @@ def write_global_compat_info(edt):
             if bus is not None and bus not in compat2buses[compat]:
                 compat2buses[compat].append(bus)
 
+    for compat, nodes in edt.compat2nodes.items():
+        for node in nodes:
+            if compat == "fixed-partitions":
+                for child in node.children.values():
+                    if "label" in child.props:
+                        label = child.props["label"].val
+                        macro = f"COMPAT_{str2ident(compat)}_LABEL_{str2ident(label)}"
```

Comment:
Is this meant as a good-enough-for-now solution? I am asking because I'm not sure labels are unique in general. I know non-unique labels in zephyr devicetrees break device_get_binding(), so we rely on unique labels, but I'm asking since we are getting away from requiring label properties in general.

Is it possible to use a node label like `boot_partition` instead of a label property like `mcuboot` here? Node labels are unique, after all.

Reply:
I think I have answered my own question. I am OK with labels since dtspec says the value of the property is binding-specific and upstream makes it seem like they're expected to seem unique, as pointed out by @pabigot earlier.

Reply:
Its meant as good-enough and maintaining the functionality we have today.  I'd prefer to stay away from nodelabel's here especially with the long standing use of the `label` property.

I'd be interested to see how we progress the flash map API w/regards to multiple flash's.

We can also add something to the binding (upstream) about label being unique.